api/config/v1alpha1/envoygateway_types.go

```diff
@@ -57,6 +63,47 @@ type EnvoyGatewaySpec struct {
 	Extension *Extension `json:"extension,omitempty"`
 }
 
+// EnvoyGatewayLogging defines logging for Envoy Gateway.
+type EnvoyGatewayLogging struct {
+	// Level is the logging level. If unspecified, defaults to "Info".
+	//
+	// +kubebuilder:default={system: info}
+	Level map[EnvoyGatewayLogComponent]LogLevel `json:"level,omitempty"`
+
+	// DevelopmentMode the log development mode or not for EnvoyGateway running, If unspecified, defaults to true.
+	//
+	// +kubebuilder:default=true
+	DevelopmentMode bool `json:"developmentMode,omitempty"`
```

Comment:
why is development mode needed, why not use `debug` level logging instead ?

Reply:
https://github.com/envoyproxy/gateway/blob/6dc6e1da07596f1279835060af626e8ce92b47d0/internal/logging/log.go#L66-L77

we can use different zap config

Reply:
how is this helping the user

Reply:
user can use different log encoder

```go
// NewProductionEncoderConfig returns an opinionated EncoderConfig for
// production environments.
func NewProductionEncoderConfig() zapcore.EncoderConfig {
	return zapcore.EncoderConfig{
		TimeKey:        "ts",
		LevelKey:       "level",
		NameKey:        "logger",
		CallerKey:      "caller",
		FunctionKey:    zapcore.OmitKey,
		MessageKey:     "msg",
		StacktraceKey:  "stacktrace",
		LineEnding:     zapcore.DefaultLineEnding,
		EncodeLevel:    zapcore.LowercaseLevelEncoder,
		EncodeTime:     zapcore.EpochTimeEncoder,
		EncodeDuration: zapcore.SecondsDurationEncoder,
		EncodeCaller:   zapcore.ShortCallerEncoder,
	}
}
```

```go
// NewDevelopmentEncoderConfig returns an opinionated EncoderConfig for
// development environments.
func NewDevelopmentEncoderConfig() zapcore.EncoderConfig {
	return zapcore.EncoderConfig{
		// Keys can be anything except the empty string.
		TimeKey:        "T",
		LevelKey:       "L",
		NameKey:        "N",
		CallerKey:      "C",
		FunctionKey:    zapcore.OmitKey,
		MessageKey:     "M",
		StacktraceKey:  "S",
		LineEnding:     zapcore.DefaultLineEnding,
		EncodeLevel:    zapcore.CapitalLevelEncoder,
		EncodeTime:     zapcore.ISO8601TimeEncoder,
		EncodeDuration: zapcore.StringDurationEncoder,
		EncodeCaller:   zapcore.ShortCallerEncoder,
	}
}
```

of course, we can use the NewDevelopmentEncoderConfig directly and remove developmentMode

Reply:
I will remove this, and use the development config .